macOS DPI experiments

csomor opened the pull request:
testing APIs to get the different High-Resolution Concepts under a common umbrella …

csomor committed:
Generate wxEVT_DPI_CHANGED event under Mac

csomor committed:
Implement wxWindow::GetDPI() on macOS and iOS

csomor committed:
Propagate wxEVT_DPI_CHANGED to child windows in wxOSX too

csomor committed:
Fix content scale factor returned by wxScreenDC on Mac

csomor commented:
@vadz, please at this as well to 3.1.4, it makes the DPI thing consistent

vadz commented on src/osx/cocoa/nonownedwnd.mm:
```diff
@@ -612,7 +613,43 @@ - (void)windowWillEnterFullScreen:(NSNotification *)notification
+                                         doubleValue];
+        if (newBackingScaleFactor != oldBackingScaleFactor)
+        {
+            wxSize oldDPI((int)(oldBackingScaleFactor*72.0),(int)(oldBackingScaleFactor*72.0));
+            wxSize newDPI((int)(newBackingScaleFactor*72.0),(int)(newBackingScaleFactor*72.0));
```
Should I change those to use `wxRound()` too? It seems it would be better, if only for consistency with `wxWindow::GetDPI()`?

vadz replied:
OK, I've done this and force pushed my changes here, I hope it's ok.

csomor replied:
yes, perfect, thanks

vadz committed:
Combine variable declaration and test in a single line

No real changes, make the code slightly shorter.

vadz committed:
Include wx/math.h in the file using wxRound()

Although it seems to be already included from some other header
implicitly, prefer to also include it explicitly to avoid any breakage
later.

vadz committed:
Add OSXMakeDPIFromScaleFactor() helper

No real changes, just avoid repeating wxRound(scaleFactor*72) in several
places and do it in this single function instead.
